src/include/execution_plan.hpp

```diff
@@ -33,6 +33,13 @@ struct hash<mscclpp::ChannelKey> {
            std::hash<int>()(static_cast<int>(key.dstBufferType)) ^ std::hash<int>()(static_cast<int>(key.channelType));
   }
 };
+
+template <>
+struct hash<std::pair<int, mscclpp::ChannelType>> {
+  std::size_t operator()(const std::pair<int, mscclpp::ChannelType>& key) const {
+    return std::hash<int>()(key.first) ^ std::hash<int>()(static_cast<int>(key.second));
```

Comment:
@Binyang2014 This looks unsafe as `(1, ChannelType::PROXY)` pair will have the same hash value as `(2, ChannelType::SM)`.

Reply:
Thanks, fixed